packages/relay/src/routes/user.ts

```diff
@@ -0,0 +1,55 @@
+import { Express } from 'express'
+import { DB } from 'anondb/node'
+import { Synchronizer } from '@unirep/core'
+import TwitterClient from '../singletons/TwitterClient'
+import { ethers } from 'ethers';
+import crypto from 'crypto';
+
+const STATE = crypto.randomUUID()
+
+export default (app: Express, db: DB, synchronizer: Synchronizer) => {
+    app.get('/api/login', async (req, res) => {
+        res.redirect(TwitterClient.authClient.generateAuthURL({
+            state: STATE,
+            code_challenge: "BDcX6UdIZ7uAJdWnCsaKL0G9O2rwzvu4"
+        }))
+    })
+
+    app.get('/api/callback', async (req, res) => {
+        try {
+            const { code, state } = req.query;
+            if (state !== STATE)
+                return res.status(500).json({
+                    error: "State isn't matching"
+                });
+            
+            TwitterClient.authClient.requestAccessToken(code as string)
+                .then(_ => TwitterClient.client.users.findMyUser())
+                .then(async userInfo => {
+
+                    const userId = userInfo.data?.id
+                    if (!userId) throw new Error('No userId')
```

Comment:
so you will check the twitter id exists, then return the hash of id?
I am not sure if our agreement is that we hash(id) in the frontend and pass it to backend (`hash(id)`)
but yeah we shouldn't allow user pass random `hash(id)` to signup (where id doesn't exist)